File: EfwDomain/src/br/com/fiap/coleta/entities/JanelaSla.java
package br.com.fiap.coleta.entities;

import java.io.Serializable;
import java.util.Date;

public class JanelaSla implements Serializable {

	private static final long serialVersionUID = 2000347320296365654L;
	private Long id;
	private Sla sla;
	private String descricao;
	private Date dataInicio;
	private Date dataFim;
	private String horaInicio;
	private String horaFim;
	private DiasSemanaJanelaSla diasSemana;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Sla getSla() {
		return sla;
	}
	public void setSla(Sla sla) {
		this.sla = sla;
	}
	public String getDescricao() {
		return descricao;
	}
	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}
	public Date getDataInicio() {
		return dataInicio;
	}
	public void setDataInicio(Date dataInicio) {
		this.dataInicio = dataInicio;
	}
	public Date getDataFim() {
		return dataFim;
	}
	public void setDataFim(Date dataFim) {
		this.dataFim = dataFim;
	}
	public String getHoraInicio() {
		return horaInicio;
	}
	public void setHoraInicio(String horaInicio) {
		this.horaInicio = horaInicio;
	}
	public String getHoraFim() {
		return horaFim;
	}
	public void setHoraFim(String horaFim) {
		this.horaFim = horaFim;
	}
	public DiasSemanaJanelaSla getDiasSemana() {
		return diasSemana;
	}
	public void setDiasSemana(DiasSemanaJanelaSla diasSemana) {
		this.diasSemana = diasSemana;
	}
	
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		JanelaSla other = (JanelaSla) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}
	
	
	
	
	
	
}

File: ServerRest/WebContent/app/model/JanelaSla.js
Ext.define('MONITOR.model.JanelaSla', {
	extend: 'Ext.data.Model',
	
    fields: [
        {name:'id', type:'int'},
        {name:'descricao', type:'string'},
        {name:'dataInicio', type:'date'},
        {name:'dataFim', type:'date'},
        {name:'horaInicio', type:'date'},
        {name:'horaFim', type:'date'}
    ],
    
    proxy: {
		type: 'rest',
		url: 'rest/janelaSla/',
        reader: {
            type: 'json'
        },
		writer: {
			type: 'json',
			writeAllFields: true
		}
	},
	
    hasOne: [
        {model: 'MONITOR.model.DiasSemanaJanelaSla', foreignKey: 'id', associationKey: 'diasSemana', getterName: "getDiasSemana", setterName: "setDiasSemana" },
        {model: 'MONITOR.model.Sla', foreignKey: 'sla.id', associationKey: 'sla', getterName: "getSla", setterName: "setSla" }
    ]
    
});